cloudcix_primitives/cloudinit_lxd.py

```diff
@@ -0,0 +1,225 @@
+"""
+Cloud-init configuration management for LXD containers
+Allows updating, reading, and scrubbing user-data, vendor-data, and network-config.
+"""
+# stdlib
+from typing import Dict, Tuple
+# libs
+from cloudcix.rcc import API_SUCCESS, CHANNEL_SUCCESS, comms_lxd
+# local
+from cloudcix_primitives.utils import HostErrorFormatter, LXDCommsWrapper
+
+__all__ = [
+    'update',
+    'read',
+    'scrub',
+]
+
+def update(
+        endpoint_url: str,
+        project: str,
+        container_name: str,
+        cloud_init_config: str,
+        config_type: str = 'user-data',
```

Comment:
This can be hard coded as the cloudinit file supplied by the customer will always be user-data. 


We do not have vendor-data and the management of the network-config will be a seperate primitive